spec/registry.md

```diff
@@ -227,6 +201,58 @@ MUST be multiplied by 100 for the purposes of formatting.
 > should format in a manner similar to:
 > > The total was 50%.
 
+##### Currency Formatting
+
+Implementations MAY support formatting a numeric value together with its currency
+when the option `style` is assigned the value `currency`.
+
+If an implementation supports currency formatting,
+it SHOULD define and support an implementation-defined _operand_ value
+that combines a value together with its currency
+expressed as a valid [Unicode Currency Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeCurrencyIdentifier).
+
+If an implementation supports currency formatting,
+the following options and their values are REQUIRED to be available on the function `:number`
+when `style=currency` is set:
+- `currencyDisplay`
+   - `symbol` (default)
+   - `narrowSymbol`
+   - `code`
+   - `name`
+- `currencySign`
+  - `accounting`
+  - `standard` (default)
+
+If an implementation supports currency formatting,
+it MAY make the following option available on the function `:number`
```

Comment:
I'm not sure that I understand your comment here. Do you mean that we ought to require all implementations to support an option specifying a currency code?

Reply:
No, but if you support `style=currency` then you must support the `currency` option. This says MAY.

Reply:
I don't see why such a requirement ought to be placed on all implementations that support currency formatting. I agree with @macchiati in https://github.com/unicode-org/message-format-wg/issues/838#issuecomment-2254300739 that this can be solved in user-space instead.

Reply:
It might be solved in the user space. Users are encouraged to do so. 

But my experience is that there are many applications that are mono-currency or which provide the currency contextually (at the document/record/transaction level). I see these people who are somewhere along the internationalization curve and it is a hard sell to get them to refactor their application just to format a number as a currency.

There are many runtime environments in which there is no currency value class/object/structure built-in: people have invented their own. Providing a way for these applications to work reasonably I think is important too.

But, finally, I think the key thing is that our functions/option sets should be as portable as possible. It simplifies the work for tools and editors and the learning curve of developers. Where we have optional functionality, the package should be as consistent as possible. Implementations can extend it, but those options will (we hope) be namespaced.